core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -941,6 +965,12 @@ private[spark] object SparkSubmitUtils {
       val ri = ModuleRevisionId.newInstance(mvn.groupId, mvn.artifactId, mvn.version)
       val dd = new DefaultDependencyDescriptor(ri, false, false)
       dd.addDependencyConfiguration(ivyConfName, ivyConfName + "(runtime)")
+      if (mvn.classifier.isDefined) {
```

Comment:
As far as I can tell, this looks ok to me. This seems like the right way to add an artifact with classifier, but I'm not certain.  Does the artifact show up in the `ResolveReport` at the end of `SparkSubmit.resolveMavenCoordinates`?